test/com/google/enterprise/adaptor/database/ValidatedUriTest.java

```diff
@@ -59,15 +67,17 @@ public void testBadProtocol() throws Exception {
   }
 
   @Test
-  public void testNoHost() throws Exception {
+  public void testNoHostOrPath() throws Exception {
     thrown.expect(URISyntaxException.class);
+    thrown.expectMessage("Expected authority");
     new ValidatedUri("http://");
   }
 
   @Test
-  public void testRelativeUri() throws Exception {
+  public void testNoHost() throws Exception {
     thrown.expect(URISyntaxException.class);
-    new ValidatedUri("foo/bar");
+    thrown.expectMessage("no host");
+    new ValidatedUri("http:/foo/bar");
```

Comment:
It may look weird, but that is exactly what File.toURI() returns (except file: instead of http:). I do think we should try the /// variant, tho.